### ISSUE TITLE
Not getting updated model when using push

### ISSUE THREAD

**Issue by chrissm79** [jenssegers/laravel-mongodb#228]
Not sure if this is on Eloquent's side, but when I use push to add an array of data to my model, the returned object does not include the new fields (although it is stored correctly in the DB). 

``` php
    $user = User::create([
        'email' => 'john.doe@example.com'
    ]);

    $user->push('name', ['first' => 'John', 'last' => 'Doe']);

    return $user;
```

returns 

``` json
{
    email: "john.doe@example.com",
    created_at: "2014-06-11 17:25:20",
    _id: "5398910018a9f4a314291a13"
}
```

However, manually adding it works

``` php
    $user = User::create([
        'email' => 'john.doe@example.com'
    ]);

    $user->name = ['first' => 'John', 'last' => 'Doe'];

    return $user;
```

returns

``` json
{
email: "john.doe@example.com",
created_at: "2014-06-11 17:28:51",
_id: "539891d318a9f49d14291a13",
name: {
        first: "John",
        last: "Doe"
    }
}
```

Thanks!

**Comment by nithilgeorge** [comment]
I am a newbie in Laravel MongoDB 
I would like to mention a problem here,
outline of my Mongo document should be like

{
     _id:****,
     subscriptions: [{list_id: "14Q3"},
        {list_id: "153"}],
     offers: [ { targetURL: "www.qwerty.com", title: "25% discount" }, 
	{ targetURL: "www.abcd.com", title: "55% discount" } ],
} 

I have used the following code to insert a list id (was successful).

DB::connection('mongodb')->collection('subscribers')->insert(array(
                    '_id' => $subscriber->device_id,
                    'subscriptions' => array('list_id' => $subscriber->list1_id),
                     ));

But when I try to push another list_id as next object it is showing error.
I used the following code for pushing

DB::connection('mongodb')->collection('subscribers')
->push('subscriptions', array('list_id' => $subscriber->list1_id));

I don't know whether i am right or wrong.
I need to store the data as the outline given above.
